app/models/manageiq/providers/infra_conversion_job.rb

```diff
@@ -0,0 +1,167 @@
+class ManageIQ::Providers::InfraConversionJob < Job
```

Comment:
Why is this under providers namespace?

Reply:
I'd rather we didn't use Job, but @agrare do you have opinions?

Reply:
What other options you'd recommend? @Fryguy 

What we need is a state machine type of thing to support this.

Reply:
Moved to root namespace now.

Reply:
> I'd rather we didn't use Job

AFAIK the other option for a state machine would be automate which this is trying to move away from.  I'm good with moving it to Job, its not the most intuitive but it is simpler than the alternatives and is used in a few places in core already (besides smartstate)